GoogleMapsComponents/wwwroot/js/objectManager.js

```diff
@@ -390,33 +392,68 @@
 
-                map.controls[position].push(clone);
+                // check if the control already exists
+                var controls = map.controls[position].getArray();
+                for (var i = 0; i < controls.length; i++) {
+                    if (controls[i].id === elem.id) {
+                        return;
+                    }
+                }
+
+                if (controlParents.hasOwnProperty(mapGuid) == false) {
+                    controlParents[mapGuid] = {}
+                }
+                if (controlParents[mapGuid].hasOwnProperty(elem.id) == false && !controlParents[mapGuid][elem.id]) {
+                    let parentElement = elem.parentElement
+                    controlParents[mapGuid][elem.id] = parentElement;
+                }
+                
+                elem.style.display = "block";
+                map.controls[position].push(elem);
             },
             removeControl(args) {
-                let map = mapObjects[args[0]];
+                let mapGuid = args[0];
+                let map = mapObjects[mapGuid];
                 let elem = args[2];
                 let position = getGooglePositionFromString(args[1].replace("\"", "").replace("\"", ""));
 
-                var arr = map.controls[position].getArray();
-                for (var i = 0, len = arr.length; i < len; i++) {
-                    if (arr[i].id === elem.id) {
-                        map.controls[position].removeAt(i);
+                var controls = map.controls[position].getArray();
+                for (var i = 0; i < controls.length; i++) {
+                    if (controls[i].id === elem.id) {
+                        let control = map.controls[position].removeAt(i);
+                        let parent = controlParents[mapGuid][control.id];
+                        if (parent) {
+                            parent.appendChild(control);
+                            control.style.display = "none";
+                        }
+                        delete controlParents[mapGuid][control.id]
                         return;
                     }
                 }
+
             },
             removeControls(args) {
-                let map = mapObjects[args[0]];
+                let mapGuid = args[0];
+                let map = mapObjects[mapGuid];
                 let position = getGooglePositionFromString(args[1].replace("\"", "").replace("\"", ""));
 
-                map.controls[position].clear();
+                while (map.controls[position].length > 0) {
+                    let control = map.controls[position].pop();
+                    let parent = controlParents[mapGuid][control.id];
+                    if (parent) {
+                        parent.appendChild(control);
+                        control.style.display = "none";
+                    }
+                    delete controlParents[mapGuid][control.id];
+                }
+                console.log(controlParents);
```

Comment:
No console logging in necessary

Reply:
I thought I removed all console.log. Will fix.